src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1615,8 +1617,10 @@ static void perform_stream_op_locked(grpc_exec_ctx *exec_ctx, void *stream_op,
     s->trailing_metadata_available =
         op_payload->recv_initial_metadata.trailing_metadata_available;
     if (op_payload->recv_initial_metadata.peer_string != NULL) {
-      gpr_atm_rel_store(op_payload->recv_initial_metadata.peer_string,
-                        (gpr_atm)gpr_strdup(t->peer_string));
+      char *old_peer_string = (char *)gpr_atm_full_xchg(
```

Comment:
Is this and the previous change fixing an existing memory leak?

Reply:
No, it's actually a behavior change.

Previously, the transport could assume that the peer string was not set, so it could just set it, without worrying about overwriting anything.  However, now that we're retrying, the value may get reset with each retry, so we actually need to free the previous value when we reset it.